app/com/m3/octoparts/util/OctoMetricsImpl.scala

```diff
@@ -0,0 +1,27 @@
+package com.m3.octoparts.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.kenshoo.play.metrics.MetricsImpl
+import play.api.{ Logger, Configuration }
+import play.api.inject.ApplicationLifecycle
+
+import scala.util.control.NonFatal
+
+/**
+ * During tests, we occasionally have cases where the asynchronous shutting down of an application means that
+ * the previous registry in the singleton has not been unregistered before starting the next test.
+ *
+ * This implementation of Metrics overrides onStart to perform onStop cleanups first before proceeding with
+ * other onStart duties
+ */
+class OctoMetricsImpl(lifecycle: ApplicationLifecycle, configuration: Configuration) extends MetricsImpl(lifecycle, configuration) {
+
+  override def onStart(): ObjectMapper = {
```

Comment:
Is it possible to add a test for this, or is it too edge-casey? I suppose you'd just be testing to see if the metric was removed from the registry.

Although the comment above mentions that this is to address issues when running tests, this implementation with the override is being used in non-test code (`UtilsModule`) in place of the default `MetricsImpl`. In this case, I think that either this needs testing, or that the custom implementation should only be used in testing.